src/mono/wasm/debugger/BrowserDebugProxy/MemberReferenceResolver.cs

```diff
@@ -264,24 +256,27 @@ async Task<JObject> ResolveAsLocalOrThisMember(string name)
                 return null;
             }
 
-            async Task<JObject> ResolveAsInstanceMember(ArraySegment<string> parts, JObject baseObject, bool hasNullCondition)
+            async Task<JObject> ResolveAsInstanceMember(ArraySegment<string> parts, JObject baseObject, bool throwOnNullReference)
             {
-                // if the prevPart has "?" at the end: true
-                bool hasResolvedObjNullCondition = hasNullCondition;
-                bool isPrevPartNull = false;
                 JObject resolvedObject = baseObject;
-                for (int i = 0; i < parts.Count; i++)
+                // parts[0] - name of baseObject
+                for (int i = 1; i < parts.Count; i++)
                 {
-                    string partTrimmed = parts[i];
-                    if (partTrimmed.Length == 0)
+                    string part = parts[i];
+                    if (part.Length == 0)
                         return null;
 
+                    if (part[^1] == '!')
+                        part = part.Remove(part.Length - 1);
+
+                    bool hasCurrentPartNullCondition = part[^1] == '?';
```

Comment:
You can get rid of this local too. And update `throwOnNullReference = partTrimmed[^1] != '?';` after the `if (resolvedObject.IsNullValuedObject())` block.

Reply:
um this is fine too. I was doing it a little differently, but this if fine.